File: src/main_without_classes.py
from math import sqrt

# constant for the data file
FILE_PATH = 'src/active-businesses.tsv'

# constants for column indices
FORMAL_NAME_INDEX = 0
INFORMAL_NAME_INDEX = 1
ADDRESS_INDEX = 2
CITY_INDEX = 3
ZIP_CODE_INDEX = 4
CATEGORY_INDEX = 5
START_DATE_INDEX = 6
LATITUDE_INDEX = 7
LONGITUDE_INDEX = 8

# Oxy's coordinates
OXY_LATITUDE = 34.126813
OXY_LONGITUDE = -118.211904

# 1 degree longitude is about 57 miles (at 34 latitude)
DEGREES_TO_MILES = 57


def read_businesses():
    """Reads the data file.

    Returns:
        list of list of strings: A list of businesses, each a list of strings.
    """
    businesses = []
    lines = read_lines()
    line_index = 0
    while line_index < len(lines):
        line = lines[line_index]
        business = line_to_business(line)
        businesses.append(business)
        line_index += 1
    return businesses


def read_lines():
    """Reads data file into a list of strings.

    Returns:
        list of strings: List of lines in the data file.
    """
    lines = []
    with open(FILE_PATH) as fd:
        for line in fd.readlines():
            lines.append(line.strip())
    return lines


def line_to_business(line):
    """Converts a line of data into multiple fields.

    A line of data contains the following fields, separated by tabs:
    * formal name
    * informal name
    * address
    * city
    * zip code
    * category
    * start date
    * latitude
    * longitude

    Arguments:
        line (str): A line from the data file.

    Returns:
        list of strings: The different fields, as strings.
    """
    return line.split('\t')


def distance_from(business, latitude, longitude):
    """Calculates the distance to a coordinate.

    This function assumes that the latitude remains around 34 degrees.

    Arguments:
        business (list of strings): The target business.
        latitude (float): The latitude of the destination.
        longitude (float): The longitude of the destination.

    Returns:
        float: The distance from Oxy, in miles.
    """
    biz_lat = get_latitude(business)
    biz_long = get_longitude(business)
    lat_diff = biz_lat - latitude
    long_diff = biz_long - longitude
    coord_dist = sqrt( (lat_diff * lat_diff) + (long_diff * long_diff) )
    return coord_dist * DEGREES_TO_MILES


def get_latitude(business):
    """Extracts the latitude from a business.

    Arguments:
        business (list of strings): The target business.

    Returns:
        float: The latitude of the business.
    """
    return float(business[LATITUDE_INDEX])


def get_longitude(business):
    """Extracts the longitude from a business.

    Arguments:
        business (list of strings): The target business.

    Returns:
        float: The longitude of the business.
    """
    return float(business[LONGITUDE_INDEX])


def is_restaurant(business):
    """Determines if a business is categorized as a restaurant.

    Arguments:
        business (list of strings): The target business.

    Returns:
        bool: True if the business is a restaurant, False otherwise.
    """
    return business[CATEGORY_INDEX] == 'Full-service restaurants'


def is_restaurant_near_oxy(business):
    """Determines if a business is a restaurant within a mile of Oxy.

    Arguments:
        business (list of strings): The target business.

    Returns:
        bool: True if the business is a restaurant near Oxy, False otherwise.
    """
    latitude = get_latitude(business)
    longitude = get_longitude(business)
    distance = distance_from(business, OXY_LATITUDE, OXY_LONGITUDE)
    return (is_restaurant(business) and distance < 1)


def print_business_name(business):
    """Prints the name of a business.

    Use the informal name if it exists, fall back to the formal name if it doesn't.

    Arguments:
        business (list of strings): The target business.
    """
    if business[INFORMAL_NAME_INDEX] == '':
        print(business[FORMAL_NAME_INDEX])
    else:
        print(business[INFORMAL_NAME_INDEX])


def main():
    """Print the names of restaurants near Oxy 
    """
    businesses = read_businesses()
    biz_index = 0
    while biz_index < len(businesses):
        business = businesses[biz_index]
        if is_restaurant_near_oxy(business):
            print_business_name(business)
        biz_index += 1


if __name__ == '__main__':
    main()

File: src/MainWithoutClasses.java
import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;
import java.lang.Math;
import java.util.ArrayList;
import java.util.Arrays;

public class MainWithoutClasses {

    /* constant for the data file */
    public static String FILE_PATH = "src/active-businesses.tsv";

    /* constants for column indices */
    public static int FORMAL_NAME_INDEX = 0;
    public static int INFORMAL_NAME_INDEX = 1;
    public static int ADDRESS_INDEX = 2;
    public static int CITY_INDEX = 3;
    public static int ZIP_CODE_INDEX = 4;
    public static int CATEGORY_INDEX = 5;
    public static int START_DATE_INDEX = 6;
    public static int LATITUDE_INDEX = 7;
    public static int LONGITUDE_INDEX = 8;

    /* Oxy's coordinates */
    public static double OXY_LATITUDE = 34.126813;
    public static double OXY_LONGITUDE = -118.211904;

    /* 1 degree longitude is about 57 miles (at 34 latitude) */
    public static double DEGREES_TO_MILES = 57;

    /**
     * Reads
     * @return     a list of businesses, each a list of strings
     */
    public static ArrayList<ArrayList<String>> read_businesses() {
        ArrayList<ArrayList<String>> businesses = new ArrayList<ArrayList<String>>();
        ArrayList<String> lines = read_lines();
        int i = 0;
        while (i < lines.size()) {
            String line = lines.get(i);
            ArrayList<String> business = line_to_business(lines.get(i));
            businesses.add(business);
            i ++;
        }
        return businesses;
    }

    /**
     * Reads data file into a list of strings.
     *
     * @return an ArrayList of strings, each a line in the data file
     */
    public static ArrayList<String> read_lines() {
        ArrayList<String> lines = new ArrayList<String>();
        String line = "";
        try {
            BufferedReader reader = new BufferedReader(new FileReader(FILE_PATH));
            line = reader.readLine();
            while (line != null) {
                lines.add(line);
                line = reader.readLine();
            }
        } catch (IOException e) {
            System.out.println("Working Directory is " + System.getProperty("user.dir") + ".");
            System.out.println("Cannot find '" + FILE_PATH + "'; quitting.");
            System.exit(1);
        }
        return lines;
    }

    /**
     * Converts a line of data into multiple fields
     *
     * @param line a line from the data file
     * @return     the different fields, as strings
     */
    public static ArrayList<String> line_to_business(String line) {
        // this creates a list of strings from a single line of the data file
        // each string corresponds to a field - formal name, informal name, address, etc.
        ArrayList<String> fields = new ArrayList<String>(Arrays.asList(line.split("\t")));
        // we are using the list of strings to represent the business, so we just return it
        return fields;
    }

    /**
     * Calculates the distance of a coordinate from Oxy
     *
     * @param business the target business
     * @param latitude  the latitude of the destination
     * @param longitude the longitude of the destination
     * @return          the distance from Oxy, in miles
     */
    public static double distance_from(ArrayList<String> business, double latitude, double longitude) {
        double biz_lat = get_latitude(business);
        double biz_long = get_longitude(business);
        double lat_diff = biz_lat - latitude;
        double long_diff = biz_long - longitude;
        double coord_dist = java.lang.Math.sqrt((lat_diff * lat_diff) + (long_diff * long_diff));
        return coord_dist * DEGREES_TO_MILES;
    }

    /**
     * Extracts the latitude from a business
     *
     * @param business the target business
     * @return         the latitude of the business
     */
    public static double get_latitude(ArrayList<String> business) {
        double lat = Double.parseDouble(business.get(LATITUDE_INDEX));
        return lat;
    }

    /**
     * Extracts the longitude from a business
     *
     * @param business the target business
     * @return         the longitude of the business
     */
    public static double get_longitude(ArrayList<String> business) {
        double lon = Double.parseDouble(business.get(LONGITUDE_INDEX));
        return lon;
    }

    /**
     * Determines if a business is categorized as a restaurant
     *
     * @param business the target business
     * @return         true if the business is a restaurant, false otherwise
     */
    public static boolean is_restaurant(ArrayList<String> business) {
        // note: to compare Strings, DO NOT use ==
        // instead, use the .equals method
        // For example, DO NOT DO
        //
        //     if ("hello" == "world") { ... }
        //
        // Instead, do
        //
        //     if ("hello".equals("world")) { ... }
        String Businessname = business.get(CATEGORY_INDEX);
        return Businessname.equals("Full-service restaurants");
    }

    /**
     * Determines if a business is a restaurant within a mile of Oxy
     *
     * @param business the target business
     * @return         true if the business is a restaurant near Oxy, false otherwise
     */
    public static boolean is_restaurant_near_oxy(ArrayList<String> business) {
        //double latitude = get_latitude(business);
        //double longitude = get_longitude(business);
        double distance = distance_from(business, OXY_LATITUDE, OXY_LONGITUDE);
        return (is_restaurant(business) && distance < 1);
    }

    /**
     * Prints the name of a business
     *
     * @param business the target business
     */
    public static void print_business_name(ArrayList<String> business) {
        // see note in is_restaurant about comparing Strings
        //
        if (business.get(INFORMAL_NAME_INDEX).equals("")){
        System.out.println(business.get(FORMAL_NAME_INDEX));
        }else if (!business.get(INFORMAL_NAME_INDEX).equals("")) {
        System.out.println(business.get(INFORMAL_NAME_INDEX));}
    }

    /**
     * Print the names of restaurants near Oxy
     */
    public static void main(String[] args) {
        ArrayList<ArrayList<String>> businesses = read_businesses();
        int biz_index = 0;
        while (biz_index < businesses.size()) {
            ArrayList<String> business = businesses.get(biz_index);
            if (is_restaurant_near_oxy(business)) {
                print_business_name(business);
            }
            biz_index ++;
        }
    }
}